lib/blocktree/blocktree.go

```diff
@@ -130,6 +130,64 @@ func (bt *BlockTree) GetAllBlocksAtNumber(hash common.Hash) (hashes []common.Has
 	return bt.root.getNodesWithNumber(number, hashes)
 }
 
+var ErrNilBlockInRange = errors.New("nil block in range")
+
+// SubBlockchain will return all the blocks between the start and
+// end hash inclusive.
+// If the end hash does not exist in the blocktree than an error
+// will be returned.
```

Comment:
supernit no need to use the future 

```suggestion
// is returned.
```